web/views.py

```
@@ -757,6 +762,65 @@
+                'user_attempt': user_attempt,
+            })
+        
+        session_data.append({
+            'session': session,
+            'exams': session_exam_data,
+        })
+    
+
+    student_analytics = None
+    if is_teacher:
+        student_analytics = get_student_analytics_data(course)
```

Ruff (0.8.2)

Undefined name `get_student_analytics_data`

(F821)